nohup: ignoring input
[2023-11-14 07:01:44,930] INFO Registered kafka:type=kafka.Log4jController MBean (kafka.utils.Log4jControllerRegistration$)
[2023-11-14 07:01:45,521] INFO Setting -D jdk.tls.rejectClientInitiatedRenegotiation=true to disable client-initiated TLS renegotiation (org.apache.zookeeper.common.X509Util)
[2023-11-14 07:01:45,809] INFO Registered signal handlers for TERM, INT, HUP (org.apache.kafka.common.utils.LoggingSignalHandler)
[2023-11-14 07:01:45,813] INFO [ControllerServer id=1] Starting controller (kafka.server.ControllerServer)
[2023-11-14 07:01:45,885] INFO authorizerStart completed for endpoint CONTROLLER. Endpoint is now READY. (org.apache.kafka.server.network.EndpointReadyFutures)
[2023-11-14 07:01:46,597] INFO Updated connection-accept-rate max connection creation rate to 2147483647 (kafka.network.ConnectionQuotas)
[2023-11-14 07:01:46,683] INFO [SocketServer listenerType=CONTROLLER, nodeId=1] Created data-plane acceptor and processors for endpoint : ListenerName(CONTROLLER) (kafka.network.SocketServer)
[2023-11-14 07:01:46,686] INFO [SharedServer id=1] Starting SharedServer (kafka.server.SharedServer)
[2023-11-14 07:01:46,790] INFO [LogLoader partition=__cluster_metadata-0, dir=/usr/local/kafka/kraft-combined-logs] Recovering unflushed segment 0. 0/1 recovered for __cluster_metadata-0. (kafka.log.LogLoader)
[2023-11-14 07:01:46,793] INFO [LogLoader partition=__cluster_metadata-0, dir=/usr/local/kafka/kraft-combined-logs] Loading producer state till offset 0 with message format version 2 (kafka.log.UnifiedLog$)
[2023-11-14 07:01:46,795] INFO [LogLoader partition=__cluster_metadata-0, dir=/usr/local/kafka/kraft-combined-logs] Reloading from producer snapshot and rebuilding producer state from offset 0 (kafka.log.UnifiedLog$)
[2023-11-14 07:01:46,795] INFO Deleted producer state snapshot /usr/local/kafka/kraft-combined-logs/__cluster_metadata-0/00000000000000000711.snapshot (org.apache.kafka.storage.internals.log.SnapshotFile)
[2023-11-14 07:01:46,795] INFO [LogLoader partition=__cluster_metadata-0, dir=/usr/local/kafka/kraft-combined-logs] Producer state recovery took 0ms for snapshot load and 0ms for segment recovery from offset 0 (kafka.log.UnifiedLog$)
[2023-11-14 07:01:46,882] WARN Found invalid messages in log segment /usr/local/kafka/kraft-combined-logs/__cluster_metadata-0/00000000000000000000.log at byte offset 51354: Record is corrupt (stored crc = 3007819699, computed crc = 2276104352). null (kafka.log.LogSegment)
[2023-11-14 07:01:46,888] INFO [ProducerStateManager partition=__cluster_metadata-0]Wrote producer snapshot at offset 711 with 0 producer ids in 2 ms. (org.apache.kafka.storage.internals.log.ProducerStateManager)
[2023-11-14 07:01:46,898] WARN [LogLoader partition=__cluster_metadata-0, dir=/usr/local/kafka/kraft-combined-logs] Corruption found in segment 0, truncating to offset 711 (kafka.log.LogLoader)
[2023-11-14 07:01:46,915] INFO [LogLoader partition=__cluster_metadata-0, dir=/usr/local/kafka/kraft-combined-logs] Loading producer state till offset 711 with message format version 2 (kafka.log.UnifiedLog$)
[2023-11-14 07:01:46,915] INFO [LogLoader partition=__cluster_metadata-0, dir=/usr/local/kafka/kraft-combined-logs] Reloading from producer snapshot and rebuilding producer state from offset 711 (kafka.log.UnifiedLog$)
[2023-11-14 07:01:46,915] INFO [ProducerStateManager partition=__cluster_metadata-0]Loading producer state from snapshot file 'SnapshotFile(offset=711, file=/usr/local/kafka/kraft-combined-logs/__cluster_metadata-0/00000000000000000711.snapshot)' (org.apache.kafka.storage.internals.log.ProducerStateManager)
[2023-11-14 07:01:46,919] INFO [LogLoader partition=__cluster_metadata-0, dir=/usr/local/kafka/kraft-combined-logs] Producer state recovery took 3ms for snapshot load and 1ms for segment recovery from offset 711 (kafka.log.UnifiedLog$)
[2023-11-14 07:01:46,975] INFO Initialized snapshots with IDs SortedSet() from /usr/local/kafka/kraft-combined-logs/__cluster_metadata-0 (kafka.raft.KafkaMetadataLog$)
[2023-11-14 07:01:47,001] INFO [raft-expiration-reaper]: Starting (kafka.raft.TimingWheelExpirationService$ExpiredOperationReaper)
[2023-11-14 07:01:47,370] INFO [RaftManager id=1] Completed transition to ResignedState(localId=1, epoch=4, voters=[1], electionTimeoutMs=1499, unackedVoters=[], preferredSuccessors=[]) from null (org.apache.kafka.raft.QuorumState)
[2023-11-14 07:01:47,379] INFO [RaftManager id=1] Completed transition to CandidateState(localId=1, epoch=5, retries=1, voteStates={1=GRANTED}, highWatermark=Optional.empty, electionTimeoutMs=1574) from ResignedState(localId=1, epoch=4, voters=[1], electionTimeoutMs=1499, unackedVoters=[], preferredSuccessors=[]) (org.apache.kafka.raft.QuorumState)
[2023-11-14 07:01:47,392] INFO [RaftManager id=1] Completed transition to Leader(localId=1, epoch=5, epochStartOffset=711, highWatermark=Optional.empty, voterStates={1=ReplicaState(nodeId=1, endOffset=Optional.empty, lastFetchTimestamp=-1, lastCaughtUpTimestamp=-1, hasAcknowledgedLeader=true)}) from CandidateState(localId=1, epoch=5, retries=1, voteStates={1=GRANTED}, highWatermark=Optional.empty, electionTimeoutMs=1574) (org.apache.kafka.raft.QuorumState)
[2023-11-14 07:01:47,421] INFO [kafka-1-raft-outbound-request-thread]: Starting (kafka.raft.RaftSendThread)
[2023-11-14 07:01:47,438] INFO [kafka-1-raft-io-thread]: Starting (kafka.raft.KafkaRaftManager$RaftIoThread)
[2023-11-14 07:01:47,511] INFO [MetadataLoader id=1] initializeNewPublishers: The loader is still catching up because we have loaded up to offset -1, but the high water mark is 712 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,507] INFO [RaftManager id=1] High watermark set to LogOffsetMetadata(offset=712, metadata=Optional[(segmentBaseOffset=0,relativePositionInSegment=51445)]) for the first time for epoch 5 based on indexOfHw 0 and voters [ReplicaState(nodeId=1, endOffset=Optional[LogOffsetMetadata(offset=712, metadata=Optional[(segmentBaseOffset=0,relativePositionInSegment=51445)])], lastFetchTimestamp=-1, lastCaughtUpTimestamp=-1, hasAcknowledgedLeader=true)] (org.apache.kafka.raft.LeaderState)
[2023-11-14 07:01:47,517] INFO [ControllerServer id=1] Waiting for controller quorum voters future (kafka.server.ControllerServer)
[2023-11-14 07:01:47,518] INFO [ControllerServer id=1] Finished waiting for controller quorum voters future (kafka.server.ControllerServer)
[2023-11-14 07:01:47,559] INFO [RaftManager id=1] Registered the listener org.apache.kafka.image.loader.MetadataLoader@722934906 (org.apache.kafka.raft.KafkaRaftClient)
[2023-11-14 07:01:47,582] INFO [MetadataLoader id=1] maybePublishMetadata(LOG_DELTA): The loader is still catching up because we have loaded up to offset 0, but the high water mark is 712 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,607] INFO [RaftManager id=1] Registered the listener org.apache.kafka.controller.QuorumController$QuorumMetaLogListener@43046521 (org.apache.kafka.raft.KafkaRaftClient)
[2023-11-14 07:01:47,674] INFO [controller-1-ThrottledChannelReaper-Fetch]: Starting (kafka.server.ClientQuotaManager$ThrottledChannelReaper)
[2023-11-14 07:01:47,686] INFO [controller-1-ThrottledChannelReaper-Produce]: Starting (kafka.server.ClientQuotaManager$ThrottledChannelReaper)
[2023-11-14 07:01:47,698] INFO [controller-1-ThrottledChannelReaper-Request]: Starting (kafka.server.ClientQuotaManager$ThrottledChannelReaper)
[2023-11-14 07:01:47,702] INFO [controller-1-ThrottledChannelReaper-ControllerMutation]: Starting (kafka.server.ClientQuotaManager$ThrottledChannelReaper)
[2023-11-14 07:01:47,803] INFO [MetadataLoader id=1] maybePublishMetadata(LOG_DELTA): The loader finished catching up to the current high water mark of 712 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,826] INFO [ExpirationReaper-1-AlterAcls]: Starting (kafka.server.DelayedOperationPurgatory$ExpiredOperationReaper)
[2023-11-14 07:01:47,849] INFO [MetadataLoader id=1] InitializeNewPublishers: initializing SnapshotGenerator with a snapshot at offset 711 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,854] INFO [MetadataLoader id=1] InitializeNewPublishers: initializing FeaturesPublisher with a snapshot at offset 711 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,854] INFO [MetadataLoader id=1] InitializeNewPublishers: initializing DynamicConfigPublisher controller id=1 with a snapshot at offset 711 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,855] INFO [MetadataLoader id=1] InitializeNewPublishers: initializing DynamicClientQuotaPublisher controller id=1 with a snapshot at offset 711 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,856] INFO [MetadataLoader id=1] InitializeNewPublishers: initializing ScramPublisher controller id=1 with a snapshot at offset 711 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,857] INFO [MetadataLoader id=1] InitializeNewPublishers: initializing DelegationTokenPublisher controller id=1 with a snapshot at offset 711 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,858] INFO [MetadataLoader id=1] InitializeNewPublishers: initializing ControllerMetadataMetricsPublisher with a snapshot at offset 711 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,859] INFO [MetadataLoader id=1] InitializeNewPublishers: initializing AclPublisher controller id=1 with a snapshot at offset 711 (org.apache.kafka.image.loader.MetadataLoader)
[2023-11-14 07:01:47,871] INFO [ControllerServer id=1] Waiting for the controller metadata publishers to be installed (kafka.server.ControllerServer)
[2023-11-14 07:01:47,871] INFO [ControllerServer id=1] Finished waiting for the controller metadata publishers to be installed (kafka.server.ControllerServer)
[2023-11-14 07:01:47,871] INFO [SocketServer listenerType=CONTROLLER, nodeId=1] Enabling request processing. (kafka.network.SocketServer)
[2023-11-14 07:01:47,890] ERROR Unable to start acceptor for ListenerName(CONTROLLER) (kafka.network.DataPlaneAcceptor)
org.apache.kafka.common.KafkaException: Socket server failed to bind to 0.0.0.0:9093: Address already in use.
	at kafka.network.Acceptor.openServerSocket(SocketServer.scala:734)
	at kafka.network.Acceptor.liftedTree1$1(SocketServer.scala:637)
	at kafka.network.Acceptor.start(SocketServer.scala:632)
	at kafka.network.SocketServer.$anonfun$enableRequestProcessing$2(SocketServer.scala:222)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:883)
	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2257)
	at kafka.network.SocketServer.chainAcceptorFuture$1(SocketServer.scala:215)
	at kafka.network.SocketServer.$anonfun$enableRequestProcessing$5(SocketServer.scala:229)
	at java.base/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4770)
	at kafka.network.SocketServer.enableRequestProcessing(SocketServer.scala:229)
	at kafka.server.ControllerServer.startup(ControllerServer.scala:389)
	at kafka.server.KafkaRaftServer.$anonfun$startup$1(KafkaRaftServer.scala:95)
	at kafka.server.KafkaRaftServer.$anonfun$startup$1$adapted(KafkaRaftServer.scala:95)
	at scala.Option.foreach(Option.scala:437)
	at kafka.server.KafkaRaftServer.startup(KafkaRaftServer.scala:95)
	at kafka.Kafka$.main(Kafka.scala:113)
	at kafka.Kafka.main(Kafka.scala)
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:459)
	at java.base/sun.nio.ch.Net.bind(Net.java:448)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
	at kafka.network.Acceptor.openServerSocket(SocketServer.scala:730)
	... 17 more
[2023-11-14 07:01:47,900] INFO [ControllerServer id=1] Waiting for all of the authorizer futures to be completed (kafka.server.ControllerServer)
[2023-11-14 07:01:47,900] INFO [ControllerServer id=1] Finished waiting for all of the authorizer futures to be completed (kafka.server.ControllerServer)
[2023-11-14 07:01:47,900] INFO [ControllerServer id=1] Waiting for all of the SocketServer Acceptors to be started (kafka.server.ControllerServer)
[2023-11-14 07:01:47,900] ERROR [ControllerServer id=1] Received a fatal error while waiting for all of the SocketServer Acceptors to be started (kafka.server.ControllerServer)
java.lang.RuntimeException: Unable to start acceptor for ListenerName(CONTROLLER)
	at kafka.network.Acceptor.liftedTree1$1(SocketServer.scala:652)
	at kafka.network.Acceptor.start(SocketServer.scala:632)
	at kafka.network.SocketServer.$anonfun$enableRequestProcessing$2(SocketServer.scala:222)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:883)
	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2257)
	at kafka.network.SocketServer.chainAcceptorFuture$1(SocketServer.scala:215)
	at kafka.network.SocketServer.$anonfun$enableRequestProcessing$5(SocketServer.scala:229)
	at java.base/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4770)
	at kafka.network.SocketServer.enableRequestProcessing(SocketServer.scala:229)
	at kafka.server.ControllerServer.startup(ControllerServer.scala:389)
	at kafka.server.KafkaRaftServer.$anonfun$startup$1(KafkaRaftServer.scala:95)
	at kafka.server.KafkaRaftServer.$anonfun$startup$1$adapted(KafkaRaftServer.scala:95)
	at scala.Option.foreach(Option.scala:437)
	at kafka.server.KafkaRaftServer.startup(KafkaRaftServer.scala:95)
	at kafka.Kafka$.main(Kafka.scala:113)
	at kafka.Kafka.main(Kafka.scala)
Caused by: org.apache.kafka.common.KafkaException: Socket server failed to bind to 0.0.0.0:9093: Address already in use.
	at kafka.network.Acceptor.openServerSocket(SocketServer.scala:734)
	at kafka.network.Acceptor.liftedTree1$1(SocketServer.scala:637)
	... 16 more
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:459)
	at java.base/sun.nio.ch.Net.bind(Net.java:448)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
	at kafka.network.Acceptor.openServerSocket(SocketServer.scala:730)
	... 17 more
[2023-11-14 07:01:47,901] ERROR Encountered fatal fault: caught exception (org.apache.kafka.server.fault.ProcessTerminatingFaultHandler)
java.lang.RuntimeException: Received a fatal error while waiting for all of the SocketServer Acceptors to be started
	at org.apache.kafka.server.util.FutureUtils.waitWithLogging(FutureUtils.java:68)
	at kafka.server.ControllerServer.startup(ControllerServer.scala:399)
	at kafka.server.KafkaRaftServer.$anonfun$startup$1(KafkaRaftServer.scala:95)
	at kafka.server.KafkaRaftServer.$anonfun$startup$1$adapted(KafkaRaftServer.scala:95)
	at scala.Option.foreach(Option.scala:437)
	at kafka.server.KafkaRaftServer.startup(KafkaRaftServer.scala:95)
	at kafka.Kafka$.main(Kafka.scala:113)
	at kafka.Kafka.main(Kafka.scala)
Caused by: java.lang.RuntimeException: Unable to start acceptor for ListenerName(CONTROLLER)
	at kafka.network.Acceptor.liftedTree1$1(SocketServer.scala:652)
	at kafka.network.Acceptor.start(SocketServer.scala:632)
	at kafka.network.SocketServer.$anonfun$enableRequestProcessing$2(SocketServer.scala:222)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:883)
	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2257)
	at kafka.network.SocketServer.chainAcceptorFuture$1(SocketServer.scala:215)
	at kafka.network.SocketServer.$anonfun$enableRequestProcessing$5(SocketServer.scala:229)
	at java.base/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4770)
	at kafka.network.SocketServer.enableRequestProcessing(SocketServer.scala:229)
	at kafka.server.ControllerServer.startup(ControllerServer.scala:389)
	... 6 more
Caused by: org.apache.kafka.common.KafkaException: Socket server failed to bind to 0.0.0.0:9093: Address already in use.
	at kafka.network.Acceptor.openServerSocket(SocketServer.scala:734)
	at kafka.network.Acceptor.liftedTree1$1(SocketServer.scala:637)
	... 16 more
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:459)
	at java.base/sun.nio.ch.Net.bind(Net.java:448)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
	at kafka.network.Acceptor.openServerSocket(SocketServer.scala:730)
	... 17 more
